qiime2/sdk/util.py

```diff
@@ -92,3 +92,29 @@ def parse_format(format_str):
     except KeyError:
         raise TypeError("No format: %s" % format_str)
     return format_record.format
+
+
+def artifact_actions(string):
+    """Plugins and actions that have as input the artifact type (string)
+
+    Parameters
+    ----------
+    string : str
+        QIIME2 artifact type
+
+    Returns
+    -------
+    list of tuples: [(q2.plugin, [q2.actions, ...]), ...]
+    """
+    commands = []
+
+    if string is not None:
+        pm = qiime2.sdk.PluginManager()
+        for pgn, pg in pm.plugins.items():
+            actions = list({a for an, a in pg.actions.items()
+                            for iname, i in a.signature.inputs.items()
+                            if str(i.qiime_type) == string})
```

Comment:
I think right here you want the condition to be:
```python
if i.qiime_type <= query_type
```
Which will be important for union-types, since their strings won't match, but they have the subtyping relationship.

Reply:
edit: operator was backwards, fixed in above comment